app/src/main/java/org/dhis2/utils/ValueUtils.kt

```diff
@@ -0,0 +1,125 @@
+                            .blockingIsEmpty()
+                    ) {
+                        val orgUnitName = d2.organisationUnitModule().organisationUnits()
+                            .byUid().eq(value)
+                            .one().blockingGet()!!.displayName()!!
+                        teAttrValue = orgUnitName
+                    }
+                }
+                ValueType.DATE, ValueType.AGE -> {
+                    teAttrValue = formatDate(teAttrValue)
+                }
+                ValueType.DATETIME -> {
+                    teAttrValue = formatDateTime(teAttrValue)
+                }
+
+                ValueType.PERCENTAGE -> {
+                    teAttrValue?.let {
+                        if (it.isNotEmpty()) {
+                            teAttrValue = "$it %"
+                        }
+                    }
+                }
+                else -> {
+                    teAttrValue = transformOptionSet(optionSetUid, d2, value)
+                }
+            }
+            return teAttrValue
+        }
+
+        private fun transformOptionSet(optionSetUid: String?, d2: D2, value: String?): String? {
+            var teAttrValue = value
+            if (optionSetUid != null) {
+                val optionCode = value
+                if (optionCode != null) {
+                    val option =
+                        d2.optionModule().options().byOptionSetUid().eq(optionSetUid).byCode()
+                            .eq(optionCode).one().blockingGet()
+                    if (option != null && (option.code() == optionCode || option.name() == optionCode)) {
+                        teAttrValue = option.displayName()
+                    }
+                }
+            }
+            return teAttrValue
+        }
+
+        private fun formatDate(value: String?): String {
+            return if (value?.isNotEmpty() == true) {
+                var formattedDate = ""
+                val date = try {
+                    DateUtils.oldUiDateFormat().parse(value)
+                } catch (e: ParseException) {
+                    null
+                }
+                date?.let {
+                    formattedDate = DateUtils.uiDateFormat().format(date)
+                }
+                formattedDate
+            } else {
+                ""
+            }
+        }
+
+        private fun formatDateTime(value: String?): String {
+            return if (value?.isNotEmpty() == true) {
+                var formattedDate = ""
+                val date = try {
+                    DateUtils.databaseDateFormatNoSeconds().parse(value)
+                } catch (e: ParseException) {
+                    null
+                }
+                date?.let {
+                    formattedDate = DateUtils.uiDateTimeFormat().format(date)
+                }
+                formattedDate
+            } else {
+                ""
+            }
```

Comment:
These methods are almost duplicated code from the `getFriendlyQueryData` method in the `SearchTEIViewModel`. Maybe there it's good idea to have this formatter in one place en use it everywhere it is needed.

Reply:
Hi yes I did notice, however the getFriendlyQueryData was mapping it slightly differently, I will give it some thought and try to reuse it :D

Reply:
Updated the PR